src/common/file-url-provider.ts

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { provideBlob } from './blob-provider';
+import { WindowUtils } from './window-utils';
+
+export class FileURLProvider {
+    constructor(private windowUtils: WindowUtils) {}
```

Comment:
This needs tests?

Reply:
Agreed... you'll need to pass `provideBlob` through the constructor if you want to mock it on the test too.